pkg/gitproviders/common.go

```diff
@@ -260,11 +260,11 @@ func UploadDeployKey(owner, repoName string, deployKey []byte) error {
 func DeployKeyExists(owner, repoName string) (bool, error) {
 	return gitProviderHandler.(GitProviderHandler).DeployKeyExists(owner, repoName)
 }
-func CreatePullRequestToUserRepo(userRepRef gitprovider.UserRepositoryRef, targetBranch string, newBranch string, files []gitprovider.CommitFile, commitMessage string, prTitle string, prDescription string) error {
+func CreatePullRequestToUserRepo(userRepRef gitprovider.UserRepositoryRef, targetBranch string, newBranch string, files []gitprovider.CommitFile, commitMessage string, prTitle string, prDescription string) (string, error) {
```

Comment:
There are way too many arguments to this function. Anything more than 4 tends to get unwieldy. I realize it was set up that way before this PR, just wanted to comment.

Reply:
Given that this func returns a `string`, I don't know what the string represents. I assume a URL in this case? Can we return a golang `url.Url` type? Or possibly the pull request object that is being created?

Reply:
I can absolutely return the pr object and get the url string in add. Unfortunately I dont think I can lower the number of arguments. Gitproviders needs all of that info and I could take some of that info and generate it in common but that could limit future functionality.

Reply:
Yeah I would say that would be a separate PR to make that change.

I would use a `struct` as an argument like we do on `AddParams` to make it a little more understandable.